tests/unit/package_managers/test_general.py

```diff
@@ -132,3 +141,92 @@ def test_extract_git_info(url: str, nonstandard_info: Any) -> None:
     }
     info.update(nonstandard_info or {})
     assert general.extract_git_info(url) == info
+
+
+@pytest.mark.asyncio
+async def test_async_download_binary_file(tmp_path: Path, caplog: pytest.LogCaptureFixture) -> None:
+    url = "http://example.com/file.tar"
+    download_path = tmp_path / "file.tar"
+
+    response, session = MagicMock(), MagicMock()
+    response.content.read = MockReadChunk().read_chunk
+
+    async def mock_aenter() -> MagicMock:
+        return response
+
+    session.get().__aenter__.side_effect = mock_aenter
+
+    await _async_download_binary_file(session, url, download_path)
+
+    with open(download_path, "rb") as f:
+        assert f.read() == b"first_chunk-second_chunk-"
+
+    assert session.get.called
+    assert session.get.call_args == mock.call(url, auth=None, raise_for_status=True)
+
+
+@pytest.mark.asyncio
+async def test_async_download_binary_file_exception(
+    tmp_path: Path, caplog: pytest.LogCaptureFixture
+) -> None:
+    url = "http://example.com/file.tar"
+    download_path = tmp_path / "file.tar"
+
+    session = MagicMock()
+
+    exception_message = "This is a test exception message."
+    session.get().__aenter__.side_effect = Exception(exception_message)
+
+    with pytest.raises(FetchError) as exc_info:
+        await _async_download_binary_file(session, url, download_path)
+
+    assert f"Unsuccessful download: {url}" in caplog.text
+    assert str(exc_info.value) == f"exception_name: Exception, details: {exception_message}"
+
+
+@pytest.mark.asyncio
+@mock.patch("cachi2.core.package_managers.general._async_download_binary_file")
+async def test_async_download_files(
+    mock_download_file: MagicMock,
+    tmp_path: Path,
+    mock_async_download_binary_file: MagicMock,
+) -> None:
+    files_to_download: Dict[str, Union[str, PathLike[str]]] = {
+        "file1": str(tmp_path / "path1"),
+        "file2": str(tmp_path / "path2"),
+        "file3": str(tmp_path / "path3"),
+    }
+
+    concurrency_limit = 2
+
+    mock_download_file.return_value = mock_async_download_binary_file
+
+    await async_download_files(files_to_download, concurrency_limit)
+
+    assert mock_download_file.call_count == 3
+
+    # Assert that mock_download_file was called with the correct arguments
+    for call in mock_download_file.mock_calls:
+        # call looks like this:
+        # call(<AsyncMock name='RetryClient().__aenter__()' id='1'>, 'file1', '/tmp/path1')
+        file, path = re.findall(r"'([^']+)'", str(call))[-2:]
```

Comment:
would this work?
```suggestion
        file, path = call.args
```
or treating the `call` as a tuple https://docs.python.org/3/library/unittest.mock.html#calls-as-tuples

Reply:
Thanks! It worked.